src/main/java/homer/model/temperaturechangers/AbstractTemperatureChanger.java

```diff
@@ -0,0 +1,102 @@
+package homer.model.temperaturechangers;
+
+import java.time.Duration;
+
+import homer.api.DeviceInfo;
+import homer.common.Temperature;
+import homer.model.environment.Environment;
+
+/**
+ * Uses template method on updateTick. At a default intensity, it changes the temperature by 
+ * 1 celsius degree each hour;
+ */
+public abstract class AbstractTemperatureChanger implements TemperatureChanger {
+
+    private final double normalizer = 1 / (1000f * 3600); // used to normalize the intensity
+    private final double maxIntensity;
+    private final double minIntensity;
+    private double intensity;
+    private final Environment environment;
+    private DeviceInfo info;
+    private final Temperature minTemperature;
+    private final Temperature maxTemperature;
+
+    /**
+     * 
+     * @param minIntensity
+     * @param maxIntensity
+     * @param currentIntensity
+     * @param environment
+     * @param minTemperature
+     * @param maxTemperature
+     */
+    public AbstractTemperatureChanger(final double minIntensity, final double maxIntensity, 
+        final double currentIntensity, final Environment environment, final Temperature minTemperature, 
+        final Temperature maxTemperature) {
+        this.minIntensity = minIntensity;
+        this.maxIntensity = maxIntensity;
+        this.intensity = currentIntensity;
+        this.environment = environment;
```

Comment:
You will get a spotbugs error just like for my thermometer, since the object can be mutated outside, I don't know yet what to do about it.

Reply:
i know, but i don't really know how to fix it.